File: 2015/day1.py
with open('day1.txt') as input:
    data = input.read().strip()
    d1 = len([x for x in data if x=='('])
    print(2*d1-len(data))
    pos = 0
    cnt = 1
    for char in data:
        pos = pos + 1 if char == '(' else pos-1
        if pos < 0:
           break
        cnt += 1
    print(cnt)

# 8 minutes
# 5 minutes

File: 2021/day1.hs
module Main where

process :: [Int] -> Int
process input = length (filter (\a -> (fst a) < (snd a)) (zip input $ tail input))

threes :: [Int] -> [Int]
threes x = map (\(a,b,c)->a+b+c) (zip3 x (drop 1 x) (drop 2 x))

main = do s1 <- readFile "day1.txt"
          putStrLn $ show $ process $ map read $ lines s1
          putStrLn $ show $ process $ threes $ map read $ lines s1

